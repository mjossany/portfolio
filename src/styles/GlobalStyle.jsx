import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  /* :root{
    --primary-color: #007bff;
    --primary-color-light: #057fff;
    --secondary-color: #6c757d;
    --background-dark-color: #10121A;
    --background-dark-gray: #191D2B;
    --border-color: #2e344e;
    --background-light-color: #F1F1F1;
    --background-light-color-2: rgba(3,127,255,.3);
    --white-color: #FFF;
    --font-light-color: #a4acc4;
    --font-dark-color: #313131;
    --font-dark-color-2: #151515;
    --sidebar-dark-color: #191D2B;
    --scrollbar-bg-color: #383838;
    --scrollbar-thump-color: #6b6b6b;
    --scrollbar-track-color: #383838;
  } */

  .light-theme{
    --primary-color: #007bff;
    --primary-color-light: #057FFF;
    --secondary-color: #ff7675;
    --background-dark-color: #F1F1F1;
    --background-dark-gray: #e4e4e4;
    --border-color: #cbced8;
    --background-light-color: #F1F1F1;
    --background-light-color-2: rgba(3,127,255,.3);
    --white-color: #151515;
    --font-light-color: #313131;
    --font-dark-color: #313131;
    --font-dark-color-2: #151515;
    --sidebar-dark-color: #E4E4E4;
    --scrollbar-bg-color: #383838;
    --scrollbar-thump-color: #6b6b6b;
    --scrollbar-track-color: #383838;
    --title-span-color: #EBEBEB;
}

  .dark-theme {
    --primary-color: #007bff;
    --primary-color-light: #057fff;
    --secondary-color: #6c757d;
    --background-dark-color: #10121A;
    --background-dark-gray: #191D2B;
    --border-color: #2e344e;
    --background-light-color: #F1F1F1;
    --background-light-color-2: rgba(3,127,255,.3);
    --white-color: #FFF;
    --font-light-color: #a4acc4;
    --font-dark-color: #313131;
    --font-dark-color-2: #151515;
    --sidebar-dark-color: #191D2B;
    --scrollbar-bg-color: #383838;
    --scrollbar-thump-color: #6b6b6b;
    --scrollbar-track-color: #383838;
    --title-span-color: rgba(25, 29, 43, .44);
  }

  * {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
    list-style: none;
    text-decoration: none;
    font-family: 'Nunito', sans-serif;
    font-size: 16px;
    @media screen and (max-width: 768px) {
      font-size: 12px;
    }
  }

  body {
    background-color: var(--background-dark-color);
    color: var(--font-light-color);
  }

  body::-webkit-scrollbar {
    width: 9px;
    background-color: #383838;
  }

  body::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: #6b6b6b;
  }

  body::-webkit-scrollbar-track {
    border-radius: 10px;
    background-color: #383838;
  }

  a{
    font-family: inherit;
    color: inherit;
    font-size: inherit;
    font-size: 1rem;
  }

  h1 {
    font-size: 4rem;
    color: var(--white-color);
    span {
      font-size: 4rem;
    }
  }

  span {
    color: var(--primary-color);
  }

  h6 {
    color: var(--white-color);
    font-size: 1.2rem;
    padding-bottom: .6rem;
  }
  //Utilities
  .u-margin-bottom {
    margin-bottom: 4rem;
  }

  // Global Media Queries
  .nav-toggle {
    @media screen and (max-width: 1200px) {
      transform: translateX(0);
    }
  }
`;

export default GlobalStyle;
