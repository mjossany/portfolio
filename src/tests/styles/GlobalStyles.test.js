import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import GlobalStyle from '../../styles/GlobalStyle';

describe('GlobalStyle', () => {
  it('renders properly', () => {
    renderer.create(<GlobalStyle />);
    expect(document.head).toMatchSnapshot();
  });
});
