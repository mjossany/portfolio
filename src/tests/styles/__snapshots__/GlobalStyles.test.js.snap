// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyle renders properly 1`] = `
<head>
  <style
    data-styled="active"
    data-styled-version="5.3.3"
  />
</head>
`;
